pkg/genericcontrolplane/apiextensions.go

```diff
@@ -86,10 +86,12 @@ func createAPIExtensionsConfig(
 			SharedInformerFactory: externalInformers,
 		},
 		ExtraConfig: apiextensionsapiserver.ExtraConfig{
-			CRDRESTOptionsGetter: apiextensionsoptions.NewCRDRESTOptionsGetter(etcdOptions),
```

Comment:
can we split the discovery change and the client new func change? The later should be easier to upstream. The former will lead to a performance discussion (which we might win).

Reply:
Definitely can do. Is it ok if we defer that until later, when we decide to start creating upstream PRs?

Reply:
This commit still has far too much of different things going on. It will catch us later. I think it's worth to split the New* funcs appart into their own commit.

Reply:
Will do, thanks!